src/main/java/de/dennisguse/opentracks/services/TrackDeleteService.java

```diff
@@ -59,22 +62,25 @@ public int onStartCommand(Intent intent, int flags, int startId) {
         return START_NOT_STICKY;
     }
 
-    private void deleteTracks(@NonNull ArrayList<Track.Id> trackIds) {
-        sendResult(null,0, trackIds.size());
+    private void deleteTracks(@NonNull List<Track.Id> trackIds) {
         serviceExecutor.execute(() -> {
             ContentProviderUtils contentProviderUtils = new ContentProviderUtils(this);
-            for (int i = 0; i < trackIds.size(); i++) {
-                updateNotification(i + 1, trackIds.size());
-                contentProviderUtils.deleteTrack(this, trackIds.get(i));
-                sendResult(trackIds.get(i), i + 1, trackIds.size());
+            int rowsDeleted = contentProviderUtils.deleteTracks(this, trackIds);
+            PreferencesUtils.addTotalRowsDeleted(rowsDeleted);
+            int totalRowsDeleted = PreferencesUtils.getTotalRowsDeleted();
+            if (totalRowsDeleted > TOTAL_DELETED_ROWS_VACUUM_THRESHOLD) {
```

Comment:
We could move this decision as a static method into the CustomContentProvider (incl. the TOTAL_DELETED_ROWS_VACUUM_THRESHOLD); then everything DB related is one place and TrackDeleteService would just call it.

Reply:
Yes, that would reduce some complexity.
I did it this way, so the TrackDeleteService can update the notification that the vacuuming is in progress.
But if this info is not so important, we can simplify the code.

Reply:
Nearly 20 lines less code (and complexity) if we move this logic as proposed into the CustomContentProvider. That's not worth to keep for the notification which is hard to find anyway. I've simplified it. Thanks for the suggestion.